proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java

```diff
@@ -58,15 +58,16 @@ public void exception(Throwable t) throws Exception
             return;
         }
 
-        ServerInfo def = bungee.getServerInfo( con.getPendingConnection().getListener().getFallbackServer() );
-        if ( server.getInfo() != def )
+        ServerInfo def = con.updateAndGetNextServer( server.getInfo() );
+        ServerKickEvent event = bungee.getPluginManager().callEvent( new ServerKickEvent( con, server.getInfo(), TextComponent.fromLegacyText( bungee.getTranslation( "server_went_down" ) ), def, ServerKickEvent.State.CONNECTED, ServerKickEvent.Cause.EXCEPTION ) );
+        if ( event.isCancelled() && event.getCancelServer() != null )
```

Comment:
Don't checked in original Bungee code (https://github.com/SpigotMC/BungeeCord/blob/812141f4006cd500780f876921c2df8648a0d746/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java#L462).
And I don't think we must protect API from potentially bad usages, since in some case that can be the desired use. Maybe I'm wrong?